src/coreclr/src/vm/compile.cpp

```diff
@@ -4843,26 +4843,14 @@ static void SpecializeComparer(SString& ss, Instantiation& inst)
         CorElementType et = elemTypeHnd.GetVerifierCorElementType();
         if (et == ELEMENT_TYPE_I1 ||
             et == ELEMENT_TYPE_I2 ||
-            et == ELEMENT_TYPE_I4)
-        {
-            ss.Set(W("System.Collections.Generic.Int32EnumComparer`1"));
-            return;
-        }
-        if (et == ELEMENT_TYPE_U1 ||
+            et == ELEMENT_TYPE_I4 ||
+            et == ELEMENT_TYPE_I8 ||
+            et == ELEMENT_TYPE_U1 ||
             et == ELEMENT_TYPE_U2 ||
-            et == ELEMENT_TYPE_U4)
-        {
-            ss.Set(W("System.Collections.Generic.UInt32EnumComparer`1"));
-            return;
-        }
-        if (et == ELEMENT_TYPE_I8)
-        {
-            ss.Set(W("System.Collections.Generic.Int64EnumComparer`1"));
-            return;
-        }
-        if (et == ELEMENT_TYPE_U8)
+            et == ELEMENT_TYPE_U4 ||
+            et == ELEMENT_TYPE_U8)
         {
-            ss.Set(W("System.Collections.Generic.UInt64EnumComparer`1"));
+            ss.Set(W("System.Collections.Generic.EnumComparer`1"));
```

Comment:
`LongEnumEqualityComparer` below should get the same fix

Reply:
Good catch! PTAL again..